### DISCUSSION TITLE
Empty base class optimisation (EBCO) std::tuple

### DISCUSSION THREAD

**Discussion by jt-9** [microsoft/STL#5004] [Q&A] [answered]
Is Microsoft STL **tuple** empty class optimisation disabled or not implemented?
I have made a simple test with [tuple sizeof](https://godbolt.org/z/9ThzK78cT) and, it appears, MSVC compiler even with /O1 (small size)
does not apply empty base class optimisation in contrast to gcc and clang.
Microsoft STL **std::unique_ptr** implementation internally relies on [_Compressed_pair](https://github.com/microsoft/STL/blob/faccf0084ed9b8b58df103358174537233b178c7/stl/inc/xmemory#L1488) which supports EBCO, but not std::tuple
I am working on utility class similar to std::unique_ptr but with different purpose [link](https://github.com/jt-9/raii-unique_rc)

What would You suggest as a cross-platform standard solution to hold members efficiently? Thank you.

**Comment by StephanTLavavej** [comment, marked as answer]
Not implemented. (And we can't implement that without breaking ABI; see https://github.com/microsoft/STL/issues/197.)

`[[no_unique_address]]` on other platforms and `[[msvc::no_unique_address]]` on MSVC can avoid paying space for data members without requiring the elaborate base class scheme.

**Reply by jt-9** [reply]
Thank you.
Missed that, sorry
